### PR TITLE
Enable requesting multiple types of evidence in one message

### PR DESCRIPTION
**Pull request by bogdan-zaharia**
Link to Jira card: https://hackney.atlassian.net/browse/DOC-297

- Updated the new request form to a multipart form based on the new designs
- Fixed an issue with selecting multiple types of evidence in the new request form
- Updated the route to contain the steps of the form
- Added new components for each step of the form
- Added unit tests for the new components and fixed failing tests
- Updated cypress tests

### CODE REVIEW

File: src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        return <div>Not Found</div>;
+    }
+  }
+
+  const currentValidationSchema = schema[currentStep - 1];
+
+  const submitHandler = async (values: EvidenceRequestForm) => {
+    const payload = buildEvidenceRequestRequest(values);
+    if (currentStep < steps.length) {
+      currentStep += 1;
```
Comment (chrishutchinson) [inline src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx:107]: Editing this value seems unexpected. Either we should infer it from `router.query.step` or we should have it in state (ideally the former)?

File: src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        ...payload,
+        userRequestedBy: user.email,
+        notificationEmail: user.email,
+      };
+      await gateway.createEvidenceRequest(user.email, requestPayload);
+      setComplete(true);
+    } catch (err) {
+      setSubmitError(true);
+      setComplete(true);
+    }
```
Comment (chrishutchinson) [inline src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx:124]: You can use `finally {}` to do the setComplete action:

```tsx
try {
  // ...
} catch(err) {
  // ...
} finally {
  setComplete(true);
}
```

Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#the_finally-block

**Review by chrishutchinson — CHANGES_REQUESTED** [review]

File: src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx (hunk 1 above)
Comment (bogdan-zaharia) [reply src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx]: addressed

File: src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx (hunk 2 above)
Comment (bogdan-zaharia) [reply src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx:124]: addressed

File: src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  team,
+  user,
+}) => {
+  const { push, query, replace } = useRouter();
+  const [complete, setComplete] = useState(false);
+  const [submitError, setSubmitError] = useState(false);
+  const [deliveryMethods, setDeliveryMethods] = useState<string[]>(['']);
+  const [previousStepNumber, setPreviousStepNumber] = useState(0);
+
+  let currentStep: number = Array.isArray(query.step)
```
Comment (chrishutchinson) [inline src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx:49]:
```suggestion
  const currentStep: number = Array.isArray(query.step)
```

Comment (bogdan-zaharia) [reply src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx]: addressed

**Review by chrishutchinson — APPROVED** [review]

**Pull request merged**